### PR TITLE
fix: workaround customized local envs

### PR DESCRIPTION
**Pull request by swatDong**
Load `.env.teamsfx.local` when local debugging. This is part of custom local debug feature but needed by multi-env, so include this in go production release.

@dooriya please help review.
@KennethBWSong FYI, this is for customized local envs in samples.

### CODE REVIEW

**Comment by swatDong** [comment]
E2E TEST: https://github.com/OfficeDev/TeamsFx-UI-Test/blob/b10d19213424ab2aece2bd286521fc07cbca36f9/src/ui-test/localdebug/localdebug-tabbot.test.ts#L7

File: packages/vscode-extension/src/debug/commonUtils.ts [hunk 1]
```diff
@@ -38,6 +38,24 @@ export async function getLocalEnv(): Promise<{ [key: string]: string } | undefin
   if (isMultiEnvEnabled()) {
     // use localSettings.json as input to generate the local debug envs
     env = await getLocalDebugEnvs();
+
+    // try load .env.teamsfx.local for tab/func/bot
```
Comment (a1exwang) [inline packages/vscode-extension/src/debug/commonUtils.ts:42]: does CLI need this fix?

File: packages/vscode-extension/src/debug/commonUtils.ts [hunk 2]
```diff
@@ -38,6 +38,24 @@ export async function getLocalEnv(): Promise<{ [key: string]: string } | undefin
...
+    // TODO: remove this workaround after fully supporting custom local debug
+    env = await appendEnvWithPrefix(
```
Comment (dooriya) [inline packages/vscode-extension/src/debug/commonUtils.ts:44]: How about the CLI support for customized environment variables?

File: packages/vscode-extension/src/debug/commonUtils.ts (hunk 1 above)
Comment (swatDong) [reply packages/vscode-extension/src/debug/commonUtils.ts:42]: Good catch. I'll move this to local debug plugin.

File: packages/vscode-extension/src/debug/commonUtils.ts (hunk 2 above)
Comment (swatDong) [reply packages/vscode-extension/src/debug/commonUtils.ts:44]: Good catch. I'll move this to local debug plugin.

File: packages/vscode-extension/src/debug/commonUtils.ts (hunk 1 above)
Comment (swatDong) [reply packages/vscode-extension/src/debug/commonUtils.ts]: Moved changes to local debug plugin so it works for CLI as well. Added test case.

File: packages/vscode-extension/src/debug/commonUtils.ts (hunk 2 above)
Comment (swatDong) [reply packages/vscode-extension/src/debug/commonUtils.ts]: Moved changes to local debug plugin so it works for CLI as well. Added test case.

**Review by dooriya — DISMISSED** [review]

**Review by a1exwang — DISMISSED** [review]

**Comment by kuojianlu** [comment]
How to let user know this customization? Is it necessary to update [the help doc](https://github.com/OfficeDev/TeamsFx/blob/main/docs/fx-core/localdebug-help.md)?

**Comment by swatDong** [comment]
> How to let user know this customization? Is it necessary to update [the help doc](https://github.com/OfficeDev/TeamsFx/blob/main/docs/fx-core/localdebug-help.md)?

Updated help doc. Added section about environment variable customization.

**Review by kuojianlu — APPROVED** [review]

**Review by dooriya — APPROVED** [review]

**Pull request merged**